### PR TITLE
Pn 6205 fe import dei testi tradotti nelle webapp

### PR DESCRIPTION
**Pull request by NicolaGiornetta**
## Short description
Added new languages support.

## List of changes proposed in this pull request
- Added new translation JSONs
- Fixed some components to properly update when change languages
- Fixed a couple of tests

## How to test
Switch between languages to see effects.

### CODE REVIEW

File: packages/pn-personafisica-webapp/src/component/Contacts/LegalContactsList.tsx [hunk 1]
```diff
@@ -30,16 +30,16 @@ const LegalContactsList = ({ recipientId, legalAddresses }: Props) => {
         </Grid>
       </Grid>
     ),
-    []
+    [i18n.language]
   );
   const defaultAddress = useMemo(
     () => legalAddresses.find((a) => a.senderId === 'default' && a.pecValid === true),
-    [legalAddresses]
+    [legalAddresses, i18n.language]
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/LegalContactsList.tsx:37]: Are we sure that i18n.language dependency is needed?

File: packages/pn-personafisica-webapp/src/component/Contacts/LegalContactsList.tsx [hunk 2]
```diff
@@ -30,16 +30,16 @@ const LegalContactsList = ({ recipientId, legalAddresses }: Props) => {
...
   );
 
   const verifyingAddress = useMemo(
     () => legalAddresses.find((a) => a.senderId === 'default' && a.pecValid === false),
-    [legalAddresses]
+    [legalAddresses, i18n.language]
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/LegalContactsList.tsx:42]: as before

File: packages/pn-personafisica-webapp/src/component/Deleghe/Delegates.tsx [hunk 3]
```diff
@@ -78,8 +78,8 @@ const Delegates = () => {
       label: t('deleghe.table.status'),
       width: '18%',
       getCellLabel(value: string) {
-        const { label, color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
-        return <Chip label={label} color={color} data-testid={`statusChip-${label}`} />;
+        const { color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
+        return <Chip label={t(`deleghe.table.${value}`)} color={color} />;
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Deleghe/Delegates.tsx:82]: why did you remove the data-testid?

File: packages/pn-personafisica-webapp/src/component/Deleghe/Delegates.tsx [hunk 4]
```diff
@@ -78,8 +78,8 @@ const Delegates = () => {
       label: t('deleghe.table.status'),
       width: '18%',
       getCellLabel(value: string) {
-        const { label, color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
-        return <Chip label={label} color={color} data-testid={`statusChip-${label}`} />;
+        const { color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Deleghe/Delegates.tsx:81]: Try to move the localization into the getDelegationStatusLabelAndColor function

File: packages/pn-personafisica-webapp/src/component/Deleghe/Delegators.tsx [hunk 5]
```diff
@@ -73,7 +73,7 @@ const Delegators = () => {
       getCellLabel(value: string, row: Item) {
         const { label, color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
         if (value === DelegationStatus.ACTIVE) {
-          return <Chip label={label} color={color} data-testid={`statusChip-${label}`} />;
+          return <Chip label={t(`deleghe.table.${value}`)} color={color} data-testid={`statusChip-${label}`} />;
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Deleghe/Delegators.tsx:76]: Same as before

File: packages/pn-personafisica-webapp/src/component/Deleghe/MobileDelegates.tsx [hunk 6]
```diff
@@ -37,8 +37,8 @@ const MobileDelegates = () => {
       id: 'status',
       label: t('deleghe.table.status'),
       getLabel(value: string) {
-        const { label, color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
-        return <Chip label={label} color={color} />;
+        const { color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
+        return <Chip label={t(`deleghe.table.${value}`)} color={color} />;
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Deleghe/MobileDelegates.tsx:41]: Same as before

File: packages/pn-personafisica-webapp/src/component/Deleghe/MobileDelegators.tsx [hunk 7]
```diff
@@ -24,9 +24,9 @@ const MobileDelegators = () => {
       id: 'status',
       label: t('deleghe.table.status'),
       getLabel(value: string, row: Item) {
-        const { label, color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
+        const { color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
         if (value === DelegationStatus.ACTIVE) {
-          return <Chip label={label} color={color} />;
+          return <Chip label={t(`deleghe.table.${value}`)} color={color} />;
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Deleghe/MobileDelegators.tsx:29]: Same as before

File: packages/pn-personagiuridica-webapp/src/component/Deleghe/DelegatesByCompany.tsx [hunk 8]
```diff
@@ -100,8 +100,8 @@ const DelegatesByCompany = () => {
         width: '18%',
       },
       getValue(value: string) {
-        const { label, color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
-        return <Chip label={label} color={color} />;
+        const { color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/component/Deleghe/DelegatesByCompany.tsx:103]: Same as for PF

File: packages/pn-personagiuridica-webapp/src/component/Deleghe/DelegationsOfTheCompany.tsx [hunk 9]
```diff
@@ -170,9 +170,9 @@ const DelegationsOfTheCompany = () => {
       id: 'status',
       label: t('deleghe.table.status'),
       getValue(value: string, row: Item) {
-        const { label, color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
+        const { color } = getDelegationStatusLabelAndColor(value as DelegationStatus);
         if (value === DelegationStatus.ACTIVE) {
-          return <Chip label={label} color={color} data-testid={`statusChip-${label}`} />;
+          return <Chip label={t(`deleghe.table.${value}`)} color={color} />;
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/component/Deleghe/DelegationsOfTheCompany.tsx:175]: Same as for PF

File: packages/pn-personagiuridica-webapp/src/component/Contacts/LegalContactsList.tsx [hunk 10]
```diff
@@ -30,16 +30,16 @@ const LegalContactsList = ({ recipientId, legalAddresses }: Props) => {
         </Grid>
       </Grid>
     ),
-    []
+    [i18n.language]
   );
   const defaultAddress = useMemo(
     () => legalAddresses.find((a) => a.senderId === 'default' && a.pecValid === true),
-    [legalAddresses]
+    [legalAddresses, i18n.language]
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/component/Contacts/LegalContactsList.tsx:37]: Same as for PF

File: packages/pn-personagiuridica-webapp/src/component/Contacts/LegalContactsList.tsx [hunk 11]
```diff
@@ -30,16 +30,16 @@ const LegalContactsList = ({ recipientId, legalAddresses }: Props) => {
...
   );
 
   const verifyingAddress = useMemo(
     () => legalAddresses.find((a) => a.senderId === 'default' && a.pecValid === false),
-    [legalAddresses]
+    [legalAddresses, i18n.language]
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/component/Contacts/LegalContactsList.tsx:42]: Same as for PF

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]
@NicolaGiornetta in the login webapp the header and the footer component have localizations missed for those languages other than italian

File: packages/pn-personafisica-webapp/src/component/Deleghe/Delegates.tsx (hunk 3 above)
Comment (NicolaGiornetta) [reply packages/pn-personafisica-webapp/src/component/Deleghe/Delegates.tsx:82]: Because wasn't used in any test.

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**